Src/ILGPU/Runtime/OpenCL/CLMemoryBuffer.cs

```diff
@@ -145,14 +145,21 @@ public sealed class CLMemoryBuffer : MemoryBuffer
             int elementSize)
             : base(accelerator, length, elementSize)
         {
-            CLException.ThrowIfFailed(
-                CurrentAPI.CreateBuffer(
-                    accelerator.NativePtr,
-                    CLBufferFlags.CL_MEM_READ_WRITE,
-                    new IntPtr(LengthInBytes),
-                    IntPtr.Zero,
-                    out IntPtr resultPtr));
-            NativePtr = resultPtr;
+            if (LengthInBytes == 0)
+            {
+                NativePtr = IntPtr.Zero;
```

Comment:
I have experimented with a lot (and I really mean it 😄) hardware configurations using our OpenCL runtime. It turns out that on some ARM and AMD platforms, the `ReleaseBuffer` operation fails if its being called with a `null` pointer. If we add some extra checks to the `DisposeAcceleratorObject` method, this patch should work fine in practice with the latest drivers on all platforms.